queue/queue.go

```diff
@@ -126,31 +99,30 @@ func (qm *Manager) DeclareQueue() error {
 	if err != nil {
 		return err
 	}
-	if qm.Logger != nil {
+	if qm.logger != nil {
 		qm.LogInfo("queue declared")
 	}
-	qm.Queue = &q
+	qm.queue = &q
 	return nil
 }
 
-// ConsumeMessage is used to consume messages that are sent to the queue
+// ConsumeMessages is used to consume messages that are sent to the queue
 // Question, do we really want to ack messages that fail to be processed?
 // Perhaps the error was temporary, and we allow it to be retried?
-func (qm *Manager) ConsumeMessage(consumer, dbPass, dbURL, dbUser string, cfg *config.TemporalConfig) error {
-	db, err := database.OpenDBConnection(database.DBOptions{
-		User:     cfg.Database.Username,
-		Password: cfg.Database.Password,
-		Address:  cfg.Database.URL,
-		Port:     cfg.Database.Port,
-	})
-	if err != nil {
-		return err
-	}
-
-	// ifs the queue is using an exchange, we will need to bind the queue to the exchange
+func (qm *Manager) ConsumeMessages(ctx context.Context, wg *sync.WaitGroup, db *gorm.DB, cfg *config.TemporalConfig) error {
+	// embed database into queue manager
+	qm.db = db
```

Comment:
hm, i'd prefer the manager to not have a db and have the db be passed to each processor as needed, but this is fine for now i think

Reply:
I'm not sure what the best choice is. It's pretty much used by every single consumer for a few purposes:
1) issuing refunds
2) instantiating our model managers.

In my eyes, it's a property of the manager, so it seems like it would make sense to embed it within the manager object.

Thoughts?

If we decide it is best like that, I'll do that during my next pass-through of the queue system.

Reply:
You noted that only consumers use it, nor publishers, which is where the main issue is I guess - perhaps they shouldn’t be tied together under one class?